src/Orleans/Streams/PersistentStreams/PersistentStreamProducer.cs

```diff
@@ -50,7 +50,7 @@ public Task OnNextAsync(T item, StreamSequenceToken token)
             {
                 throw new ArgumentNullException("token", "Passing a non-null token to a non-rewindable IAsyncBatchObserver.");
             }
-            return queueAdapter.QueueMessageAsync(stream.StreamId.Guid, stream.StreamId.Namespace, item);
+            return queueAdapter.QueueMessageAsync(stream.StreamId.Guid, stream.StreamId.Namespace, item, Runtime.RequestContext.Export());
```

Comment:
Agree. After we accept this PR, I will fix the other issue as well. So next public release will have both.